### PR TITLE
chore: spliting unit tests execution per package

### PR DESCRIPTION
**Pull request by mnischay**
## Overview
spliting up the unit test execution to individual jobs to provide scaling

## Verifying Changes
check the steps below and their respective logs

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: .github/workflows/composite/action.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  using: 'composite'
+  steps:
+    - uses: actions/setup-node@v4
+      with:
+        node-version: 16
+        cache: 'npm'
+        cache-dependency-path: '**/package-lock.json'
+
+    - name: Install dependencies
+      shell: bash
```
Comment (tracy-french) [inline .github/workflows/composite/action.yml:17]: Why do you need to specify bash shell? We haven't had this before.

**Review by tracy-french — CHANGES_REQUESTED** [review]
Looks nice! Glad we could keep everything within one file. 

Could you please point the PR to `main` instead of `rc` so that we fix the issue for everyone?

File: .github/workflows/composite/action.yml (hunk 1 above)
Comment (tracy-french) [reply .github/workflows/composite/action.yml:17]: Interesting. Didn't know!

**Comment by mnischay** [comment]
@tracy-french : we will have a release mid next week so might as well release it along with all the changes ? just pointing this to main will not work given this is branch is based off of rc ... i dont think rebasing main on to this locally will work... my guess is I need to create an entirely new CR in that case

**Comment by tracy-french** [comment]
> @tracy-french : we will have a release mid next week so might as well release it along with all the changes ? just pointing this to main will not work given this is branch is based off of rc ... i dont think rebasing main on to this locally will work... my guess is I need to create an entirely new CR in that case

Cherry-pick the commit onto `main` and create a new PR.

**Pull request closed without merging**